src/h5web/visualizations/shared/TooltipMesh.tsx

```diff
@@ -1,23 +1,19 @@
 import React, { ReactElement, useCallback } from 'react';
 import { Dom, PointerEvent, useThree } from 'react-three-fiber';
 import { TooltipWithBounds, useTooltip } from '@vx/tooltip';
-import { scaleLinear } from 'd3-scale';
-import { format } from 'd3-format';
+
 import styles from './TooltipMesh.module.css';
-import { Dims } from './models';
+import { useAbscissaScale, useOrdinateScale } from './utils';
 
 type Coords = [number, number];
 
 interface Props {
-  dims: Dims;
-  data: number[][];
+  formatIndex: (t: Coords) => string;
+  formatValue: (t: Coords) => string | undefined;
 }
 
 function TooltipMesh(props: Props): ReactElement {
-  const {
-    dims: [rows, cols],
-    data,
-  } = props;
+  const { formatIndex, formatValue } = props;
```

Comment:
`TooltipMesh` now takes two functions as props: one to format the indices, one to format the value.